src/ixdat/techniques/deconvolution.py

```diff
@@ -1,249 +1,316 @@
+            # find the other parameters from the siq Molecule files
+            Molecule = plugins.siq.Molecule
+            molecule = Molecule.load(mol)
+            if D is None:
+                D = molecule.D
+                print(D)
+            if H_v_cc is None:
+                H_v_cc = 1 / (R * chip.T * molecule.H_0) * 100
+                print(H_v_cc)
+        else:
+            if D is None:
+                raise TechniqueError(
+                    "D is required to initialize ECMSImpluseResponse. Alternatively"
+                    "activate siq as plugin to automatically load D for molecule."
+                )
+            if H_v_cc is None:
+                raise TechniqueError(
+                    "H_v_cc is required to initialize ECMSImpluseResponse. Alternatively"
+                    "activate siq as plugin to automatically load H_v_cc for molecule."
                 )
+        # convert the mol flux to volumetric flux using T and p given by chip.
+        V_dot = n_dot * R * T / p
 
-            kernel = np.zeros(len(t_kernel))
-            for i in range(len(t_kernel)):
-                kernel[i] = invertlaplace(fs, tdiff[i], method="talbot")
-                print(tdiff[i])
-                print(kernel[i])
+        # make a time array
+        t_kernel = np.arange(0, duration, dt)
+        t_kernel[0] = 1e-6
 
-        elif self.type == "measured":
-            kernel = self.MS_data[1]
-            t_kernel = self.MS_data[0]
+        tdiff = (
+            t_kernel * D / (working_distance**2)
+        )  # See Krempl et al, 2021. SI: Equ. (6)
 
-        if norm:
-            area = np.trapz(kernel, t_kernel)
-            kernel = kernel / area
+        def calc_H(s):
+            # See Krempl et al, 2021. SI: Equ 24 + definitions for kappa and lamda
+            # (between Equ 11 and 12)
+            kappa = (A_el * 1e-4 * working_distance) / gas_volume
 
-        if matrix:
+            lambda_ = V_dot / gas_volume * working_distance**2 / D
+
+            H = 1 / (
+                H_v_cc * (s + lambda_) / kappa * cosh(sqrt(s)) + sqrt(s) * sinh(sqrt(s))
+            )
+            return H
+
+        # now calculate the modelled impulse response: See Krempl et al, 2021. SI, page
+        # S4: "calculating the discrete values of the impulse response at a sampling
+        # frequency equal to the sampling frequency of the measured mass spectrometer
+        # signal by numerical inverse Laplace transformation with the Talbot method"
+        kernel = np.zeros(len(t_kernel))
+        for i in range(len(t_kernel)):
+            kernel[i] = invertlaplace(calc_H, tdiff[i], method="talbot")
+
+        if (
+            norm
+        ):  # normalize the kernel intensity to the total area under the ImpulseResponse
+            area = np.trapezoid(kernel, t_kernel)
+            kernel = kernel / area
+        if matrix:  # not sure what this does? remove it? I think it was used by a method
```

Comment:
If not remove, then rename, as per my comment in `from_measurement()`.
But I say, remove it if it is not in use. Remove basically everything that isn't needed for your demo script to run! Then if we need it, we can add a property of the `ECMSImpulseResponce` called `kernel_matrix` or something.